rosidl_typesupport_cpp/test/test_type_support.cpp

```diff
@@ -27,11 +27,11 @@ extern "C"
 // function still needs to be defined separately. Windows has gotten picky with its compiler
 // warnings recently.
 #if defined _WIN32 || defined __CYGWIN__
-__declspec(dllexport) const rosidl_message_type_support_t * test_message_type_support();
-__declspec(dllexport) const rosidl_service_type_support_t * test_service_type_support();
+__declspec(dllexport) const rosidl_message_type_support_t * test_message_type_support(void);
+__declspec(dllexport) const rosidl_service_type_support_t * test_service_type_support(void);
```

Comment:
As far as I know, these changes shouldn't be necessary in C++.  That is, C++ already enforces strict prototypes.  Did clang complain about this file?

Reply:
I think the same should be true for the rest of the .cpp file.  That is, I don't believe we need to change the `.cpp` file at all.